### ISSUE TITLE
indexing zero-length lists

### ISSUE THREAD

**Issue by thautwarm** [IronLanguages/ironpython2#813]
.NET throws `ArgumentOutOfRangeException` when executing the following code:

https://github.com/IronLanguages/ironpython2/blob/aa0526d4b786ad1fabbedbf91459903b8dc01ba3/Src/IronPython/Runtime/Operations/ListOfTOps.cs#L145-L147

**Comment by thautwarm** [comment]
It seems that such code is not used in IronPython 2. I found this bug when reusing the code for a Unity IL2CPP compatible .NET Python implementation.

**Comment by slozier** [comment]
> It seems that such code is not used in IronPython 2. I found this bug when reusing the code for a Unity IL2CPP compatible .NET Python implementation.

Indeed, looks like extended slicing doesn't work with `List<T>` and this code is never invoked.

Doesn't appear to be an issue with IronPython 3 (the code has changed since it split from ironpython2).
